File: benchmarks/xlang/src/sieve.py
import time
n = 10000000
t0 = time.perf_counter()
composite = bytearray(n + 1)
i = 2
while i * i <= n:
    if not composite[i]:
        j = i * i
        while j <= n:
            composite[j] = 1
            j += i
    i += 1
count = 0
for p in range(2, n + 1):
    if not composite[p]:
        count += 1
print(f"CHECK {count}")
print(f"MS {int((time.perf_counter() - t0) * 1000)}")

File: benchmarks/xlang/src/sieve.lua
local n = 10000000
local t0 = os.clock()
local composite = {}
local i = 2
while i * i <= n do
  if not composite[i] then
    for j = i * i, n, i do composite[j] = true end
  end
  i = i + 1
end
local count = 0
for p = 2, n do if not composite[p] then count = count + 1 end end
print("CHECK " .. count)
print("MS " .. math.floor((os.clock() - t0) * 1000))
